pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/blocks/TransferableBlock.java

```diff
@@ -167,8 +167,13 @@ public boolean isSuccessfulEndOfStreamBlock() {
   /**
    * @return whether this block represents a NOOP block
    */
+  @Deprecated(forRemoval = true)
   public boolean isNoOpBlock() {
-    return isType(MetadataBlock.MetadataBlockType.NOOP);
+    return false;
```

Comment:
This code won't compile in Java 8 (`Deprecated.forRemoval` was added in Java 8), but we should remove this method before merging this.

The reason to keep it is because there are a couple of test that still use it.